packages/@react-spectrum/table/src/TableView.tsx

```diff
@@ -513,8 +525,9 @@ function TableColumnHeader(props) {
 }
 
 let _TableColumnHeaderButton = (props, ref: FocusableRef<HTMLDivElement>) => {
+  let {isEmpty} = useTableContext();
```

Comment:
if the table is empty, maybe we just render a TableColumnHeader instead of one with a menu?

Reply:
Seems like we need to render the TableResizeColumnHeader if it is resizable, otherwise flipping from empty to non-empty doesn't replace the in view columns with resizable columns (caching probably?)